crates/uv-resolver/src/resolver/locals.rs

```diff
@@ -24,28 +24,23 @@ impl Locals {
 
         // Add all direct requirements and constraints. There's no need to look for conflicts,
         // since conflicting versions will be tracked upstream.
-        for requirement in manifest
-            .requirements
-            .iter()
-            .filter(|requirement| requirement.evaluate_markers(markers, &[]))
-            .chain(
-                manifest
-                    .constraints
-                    .iter()
-                    .filter(|requirement| requirement.evaluate_markers(markers, &[])),
-            )
-            .chain(manifest.editables.iter().flat_map(|(editable, metadata)| {
-                metadata
-                    .requires_dist
-                    .iter()
-                    .filter(|requirement| requirement.evaluate_markers(markers, &editable.extras))
-            }))
-            .chain(
-                manifest
-                    .overrides
-                    .iter()
-                    .filter(|requirement| requirement.evaluate_markers(markers, &[])),
-            )
+        for requirement in
+            manifest
+                .requirements
+                .iter()
+                .chain(manifest.constraints.iter())
+                .chain(manifest.overrides.iter())
+                .filter(|requirement| requirement.evaluate_markers(markers, &[]))
+                .chain(manifest.lookaheads.iter().flat_map(|lookahead| {
+                    lookahead.requirements().iter().filter(|requirement| {
+                        requirement.evaluate_markers(markers, lookahead.extras())
+                    })
+                }))
+                .chain(manifest.editables.iter().flat_map(|(editable, metadata)| {
+                    metadata.requires_dist.iter().filter(|requirement| {
+                        requirement.evaluate_markers(markers, &editable.extras)
+                    })
+                }))
```

Comment:
Going to DRY this up in a separate PR (it's already repeated excessively on `main`).